File: tools/code-parser-sidecar/fixtures/go-sample/service.go
package sample

import "strings"

type User struct {
	Name string
}

type UserService struct {
	users []User
}

func (s *UserService) AddUser(u User) {
	s.Validate(u)
	s.users = append(s.users, u)
}

func (s *UserService) Validate(u User) bool {
	return FormatName(u.Name) != ""
}

func FormatName(name string) string {
	return strings.ToLower(strings.TrimSpace(name))
}

func CreateDefaultService() *UserService {
	svc := &UserService{}
	svc.AddUser(User{Name: "alkut"})
	return svc
}

File: tools/code-parser-sidecar/fixtures/rust-sample/service.rs
struct User { name: String }
struct UserService { users: Vec<User> }
impl UserService {
    fn add_user(&mut self, u: User) {
        self.validate(&u);
        self.users.push(u);
    }
    fn validate(&self, u: &User) -> bool {
        format_name(&u.name) != ""
    }
}
fn format_name(name: &str) -> String { name.trim().to_lowercase() }
fn create_default_service() -> UserService {
    let mut svc = UserService { users: vec![] };
    svc.add_user(User { name: "alkut".to_string() });
    svc
}
